sequence.generator/src/test/java/com/vitreoussoftware/bioinformatics/sequence/generator/SequenceGeneratorTest.java

```diff
@@ -0,0 +1,197 @@
+                .lengthDistribution(() -> 10)
+                .encodingScheme(AcceptUnknownDnaEncodingScheme.instance)
+                .basePairDistribution(defaultDistribution)
+                .build();
+    }
+
+    /**
+     * Attempting to create without an encoding scheme fails
+     */
+    @Test(expected = NullPointerException.class)
+    public void testCreateNoEncodingScheme() {
+        SequenceGenerator.builder()
+                .lengthDistribution(() -> 10)
+                .basePairDistribution(defaultDistribution)
+                .build();
+    }
+
+
+    /**
+     * Attempting to create without an encoding scheme fails
+     */
+    @Test(expected = NullPointerException.class)
+    public void testCreateNoLengthDistribution() {
+        SequenceGenerator.builder()
+                .encodingScheme(AcceptUnknownDnaEncodingScheme.instance)
+                .basePairDistribution(defaultDistribution)
+                .build();
+    }
+
+
+    /**
+     * Attempting to create without an encoding scheme fails
+     */
+    @Test(expected = NullPointerException.class)
+    public void testCreateNoBasePairDistribution() {
+        SequenceGenerator.builder()
+                .lengthDistribution(() -> 10)
+                .encodingScheme(AcceptUnknownDnaEncodingScheme.instance)
+                .build();
+    }
+
+    /**
+     * If the length distribution keeps returning 0 we can't create a sequence and should fail.
+     */
+    @Test(expected = IllegalArgumentException.class)
+    public void testSampleFailLength0() {
+        val generator = SequenceGenerator.builder()
+                .lengthDistribution(() -> 0)
+                .encodingScheme(AcceptUnknownDnaEncodingScheme.instance)
+                .basePairDistribution(defaultDistribution)
+                .build();
+
+        generator.sample();
+    }
+
+    /**
+     * We can create a Random sequence generator for length 1
+     */
+    @Test
+    public void testSampleLength1() {
+        val length = 1;
+        val generator = SequenceGenerator.builder()
+                .lengthDistribution(() -> length)
+                .encodingScheme(AcceptUnknownDnaEncodingScheme.instance)
+                .basePairDistribution(defaultDistribution)
+                .build();
+
+        assertThat(generator.sample().length(), is(length));
+    }
+
+    /**
+     * We can create a Random sequence generator for length 10
+     */
+    @Test
+    public void testSampleLength10() {
+        val length = 10;
+        val generator = SequenceGenerator.builder()
+                .lengthDistribution(() -> length)
+                .encodingScheme(AcceptUnknownDnaEncodingScheme.instance)
+                .basePairDistribution(defaultDistribution)
+                .build();
+
+        assertThat(generator.sample().length(), is(length));
+    }
+
+
+    /**
+     * Test that we can generate a sequence from an {@link EnumeratedDistribution} containing only one element
+     */
+    @Test
```

Comment:
As with previous A/T tests this one and the subsequent one share nearly all code differing only on the base pair and could be generalized if desired.